backend/app/controllers/products.js

```diff
@@ -1470,15 +1467,24 @@ function* checkProductData(req) {
             );
         }
     }
-
-    //TODO: Add check here to point to survey service
-    // if (req.body.surveyId) {
-    //     var isExistSurvey = yield thunkQuery(Survey.select().where(Survey.id.equals(req.body.surveyId)));
-    //     if (!_.first(isExistSurvey)) {
-    //         throw new HttpError(403, 'Survey with id = ' + req.body.surveyId + ' does not exist');
-    //     }
-    // }
-
+    //Check survey service is the survey exist in there
+    const path = 'surveys/';
+
+    const requestOptions = {
+        url: config.surveyService + path + req.body.surveyId,
+        method: 'GET',
+        headers: {
+            'authorization': req.headers.authorization
+        }
+    };
+    request(
+        requestOptions,
+        function(err, response, body) {
+            if (response.statusCode != 200) {
+                throw new HttpError(response.statusCode, 'Survey with id = ' + req.body.surveyId + ' does not exist');
```

Comment:
Couldn't there be other things that cause an error here, like no auth or incorrect auth? In which case this alert would be misleading, unless I'm incorrect.

Reply:
Correct, this error should ideally be passed to an error handler. Not sure how the rest of Greyscale does it but we want to set the status code and just pass that, _maybe_ with some text if a certain error code represents something very specific at this point in the code.